### PR TITLE
Added ability to draw contours on Maps 

### PR DESCRIPTION
**Pull request by ehsteve**
This new function provides the ability to overlay contours on the plot of a map.

### CODE REVIEW

**Comment by wafels** [comment]
Can't this be achieved by using the existing composite map object?

**Comment by ehsteve** [comment]
yes, but it's clunky and weird to have to create a composite map with two copies of the same map just to be able to plot contours. This is how I've been doing it and have gotten annoyed with that process.

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -1339,6 +1339,48 @@ def draw_rectangle(self, bottom_left, width, height, axes=None, **kwargs):
...
+
+        Parameters
+        ----------
+
+        levels : `~numpy.ndarray`, list
+            A list of numbers indicating the level curves to draw;
+            e.g., to draw just the zero contour pass levels=[0]
+
+        percent : bool, default: True
+            The width of the rectangle.
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: This needs updating.

Comment (wafels) [inline sunpy/map/mapbase.py]: Should this have a figure hash too?

**Comment by Cadair** [comment]
@wafels yes this also needs a set of figure tests, and a changelog.

**Comment by ayshih** [comment]
:+1:

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -1406,6 +1406,41 @@ def draw_rectangle(self, bottom_left, width, height, axes=None, **kwargs):
 
         return [rect]
 
+    @u.quantity_input(levels=u.percent)
+    def draw_contours(self, levels, axes=None, **imshow_args):
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: `imshow_args` is not really correct, it should be `contour_args`

Comment (ehsteve) [inline sunpy/map/mapbase.py]: @Cadair fixed.

File: sunpy/map/mapbase.py [hunk 3]
```diff
@@ -1406,6 +1406,41 @@ def draw_rectangle(self, bottom_left, width, height, axes=None, **kwargs):
...
 
+    @u.quantity_input(levels=u.percent)
+    def draw_contours(self, levels, axes=None, **contour_args):
+        """
+        Draw contours of the data
+
+        Parameters
+        ----------
+
+        levels : `~astropy.units.Quantity`, list, optional
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: This says optional, which is not correct right? Also list is not correct either?

Comment (ehsteve) [inline sunpy/map/mapbase.py]: You are right. Fixed.

**Comment by Cadair** [comment]
:+1:

**Pull request merged**